libcudacxx/test/libcudacxx/std/numerics/complex.number/complex.tuple/get.pass.cpp

```diff
@@ -0,0 +1,170 @@
+    ASSERT_SAME_TYPE(T&, decltype(cuda::std::get<1>(c)));
+    static_assert(noexcept(cuda::std::get<1>(c)), "");
+    assert(i == T{28});
+  }
+  //  &&
+  {
+    cuda::std::complex<T> c{T{27}, T{28}};
+
+    auto&& r = cuda::std::get<0>(cuda::std::move(c));
+    ASSERT_SAME_TYPE(T&&, decltype(cuda::std::get<0>(cuda::std::move(c))));
+    static_assert(noexcept(cuda::std::get<0>(cuda::std::move(c))), "");
+    assert(r == T{27});
+  }
+  {
+    cuda::std::complex<T> c{T{27}, T{28}};
+
+    auto&& i = cuda::std::get<1>(cuda::std::move(c));
+    ASSERT_SAME_TYPE(T&&, decltype(cuda::std::get<1>(cuda::std::move(c))));
+    static_assert(noexcept(cuda::std::get<1>(cuda::std::move(c))), "");
+    assert(i == T{28});
+  }
+  // const &
+  {
+    const cuda::std::complex<T> c{T{27}, T{28}};
+
+    const auto& r = cuda::std::get<0>(c);
+    ASSERT_SAME_TYPE(const T&, decltype(cuda::std::get<0>(c)));
+    static_assert(noexcept(cuda::std::get<0>(c)), "");
+    assert(r == T{27});
+    const auto& i = cuda::std::get<1>(c);
+    ASSERT_SAME_TYPE(const T&, decltype(cuda::std::get<1>(c)));
+    static_assert(noexcept(cuda::std::get<1>(c)), "");
+    assert(i == T{28});
+  }
+  //  const &&
+  {
+    const cuda::std::complex<T> c{T{27}, T{28}};
+
+    const auto&& r = cuda::std::get<0>(cuda::std::move(c));
+    ASSERT_SAME_TYPE(const T&&, decltype(cuda::std::get<0>(cuda::std::move(c))));
+    static_assert(noexcept(cuda::std::get<0>(cuda::std::move(c))), "");
+    assert(r == T{27});
+  }
+  {
+    const cuda::std::complex<T> c{T{27}, T{28}};
+
+    const auto&& i = cuda::std::get<1>(cuda::std::move(c));
+    ASSERT_SAME_TYPE(const T&&, decltype(cuda::std::get<1>(cuda::std::move(c))));
+    static_assert(noexcept(cuda::std::get<1>(cuda::std::move(c))), "");
+    assert(i == T{28});
+  }
+
+#if TEST_STD_VER >= 2017
+  // `get()` allows using `complex` with structured bindings
+  {
+    cuda::std::complex<T> c{T{27}, T{28}};
+
+    auto [r, i]{c};
+    ASSERT_SAME_TYPE(T, decltype(r));
+    assert(r == T{27});
+    ASSERT_SAME_TYPE(T, decltype(i));
+    assert(i == T{28});
+  }
+  {
+    cuda::std::complex<T> c{T{27}, T{28}};
+
+    auto& [r, i]{c};
+    ASSERT_SAME_TYPE(T, decltype(r));
+    assert(r == T{27});
+    ASSERT_SAME_TYPE(T, decltype(i));
+    assert(i == T{28});
+  }
+#endif // TEST_STD_VER >= 2017
+
+  // `get()` allows using `complex` with ranges
+  // {
+  //   cuda::std::complex<T> arr[]{{T{27}, T{28}}, {T{82}, T{94}}};
+
+  //   std::same_as<cuda::std::vector<T>> decltype(auto) reals{
+  //       arr | std::views::elements<0> | cuda::std::ranges::to<cuda::std::vector<T>>()};
+  //   assert(reals.size() == 2);
+  //   assert(reals[0] == T{27});
+  //   assert(reals[1] == T{82});
+
+  //   std::same_as<std::vector<T>> decltype(auto) imags{
+  //       arr | std::views::elements<1> | cuda::std::ranges::to<cuda::std::vector<T>>()};
+  //   assert(imags.size() == 2);
+  //   assert(imags[0] == T{28});
+  //   assert(imags[1] == T{94});
+  // }
```

Comment:
i'm curious why these are commented out.

Reply:
If I'm not mistaken, ranges are not implemented yet in libcu++, am I right?

Reply:
We do have range concepts implemented from C++17 onwards, but no views

Reply:
I can implement the test and it can be enabled later when views are implemented, if you want